### ISSUE TITLE
What's the license?

### ISSUE THREAD

**Issue by pedrofaustino** [hamelsmu/AirbnbScrape#1]
hey @hamelsmu, I could not find mention of a license for the code on this repository. Could you please add one? Thanks

**Comment by hamelsmu** [comment]
Done!   Let me know if you find the code useful.

**Comment by pedrofaustino** [comment]
Thanks. Will do!

**Comment by vincent-alexander** [comment]
hey @hamelsmu, this is a really interesting project! I'm very new to Python so there is likely a silly reason for this but when I run your scraping code everything runs great except I get a "UnicodeEncodeError: 'ascii' codec can't encode character u'\xb7' in position 16: ordinal not in range(128)"  error when the writerows functions is used.  Do you know why that would be?

Thank so much for sharing this cool code!

**Comment by hamelsmu** [comment]
Can you provide more details?  Are you able to see the contents of the dictionary that is returned?  For example, if you run

MainResults = IterateMainPage('Cambridge--MA', 1)

what do you see inside the MainResults variable?

**Comment by vincent-alexander** [comment]
Hi @hamelsmu, thanks so much for responding! The error occurs when I call the writeToCSV function. The columns are written correctly but I get the error I mentioned above when the program tries to write the rows using the csv function (screen shot attached). 

I am able to see the contents of the dictionary. I have attached a screenshot of some of the dictionary results when I "print DetailResults"

![screen shot of detailresults]
(https://cloud.githubusercontent.com/assets/10539759/5759784/0d7cb4d2-9c9e-11e4-8645-2d6a3e4295f7.png)

![screenshot of writerows error](https://cloud.githubusercontent.com/assets/10539759/5759852/b3631e90-9c9e-11e4-8784-6d49bb90478c.png)

Thanks!

**Comment by hamelsmu** [comment]
Hi there,
   The code for the csv writing function is pretty simple.   It looks like
you need convert the values inside the dictionaries to UTF-8 since you are
using windows.  Why don't you try making an edit, and push your changes to
the repository.

I would take the results of the dictionary, and see if you can somehow
write those out without my function.  I am using a mac, and don't have the
same character encoding problem you are having.

I hope that helps.



On Thu, Jan 15, 2015 at 10:11 AM, vincent-alexander <
notifications@github.com> wrote:

> Hi @hamelsmu <https://github.com/hamelsmu>, thanks so much for
> responding! The error occurs when I call the writeToCSV function. The
> columns are written correctly but I get the error I mentioned above when
> the program tries to write the rows using the csv function (screen shot
> attached).
>
> I am able to see the contents of the dictionary. I have attached a
> screenshot of some of the dictionary results when I "print DetailResults"
>
> [image: screen shot of detailresults]
> <https://cloud.githubusercontent.com/assets/10539759/5759784/0d7cb4d2-9c9e-11e4-8645-2d6a3e4295f7.png>
>
> [image: screenshot of writerows error]
> <https://cloud.githubusercontent.com/assets/10539759/5759852/b3631e90-9c9e-11e4-8784-6d49bb90478c.png>
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hamelsmu/AirbnbScrape/issues/1#issuecomment-70099412>.
>



-- 

*Hamel Husain*

Linkedin <http://www.linkedin.com/in/hamelhusain> | (m) 214.893.3072

**Comment by vincent-alexander** [comment]
Thanks! I will make that change and upload the modified version

**Comment by hamelsmu** [comment]
just wanted to follow up on this, were you able to get it to work?

**Comment by nguy2261** [comment]
Hi Hamelsmu,

First, thank you for your project :+1: . It were really interesting and helpful for us to do some data mining on AirBnb for a research project :) . However, I got into the problem of unicode as mentioned above. : 

"UnicodeEncodeError: 'ascii' codec can't encode character u'\xb7' in position 16: ordinal not in range(128)" error when the writerows functions is used."

I tried many different solutions from the internet for windows 8.1 (encode, decode, utf inside the function and even in csv file). Still didn't figure out it can be solved on Windows.

Then I changed to Mac and got the same error with both PyCharm and Code Runner. So I wonder what did you use for running this project (if you remember). I would be greatly appreciate your helps. !!!
